### ISSUE TITLE
LoopProtect doesn't warn that there is an infinite loop

### ISSUE THREAD

**Issue by bugron** [FreeCodeCamp/FreeCodeCamp#10000]
LoopProtect actually protects me from an infinite it just doesn't show any relevant message.
**How that does prod site:**
![](http://i.prntscr.com/1b3790478fb8445c953995ddbd9dece5.png)
**How that does beta site:**
![](http://i.prntscr.com/80bc164a2d8c466b835a16f1df763668.png)

**Comment by raisedadead** [comment]
### :tada: :tada: :tada: 10000! :tada: :tada: :tada: 

Okay, since we are coming to LoopProtect, I think this is a good chance to have a look at an alternate better maintained library, the LP project is mostly dead with no new developments and fixes recently, and we have a more than a few open bugs on that.

**Comment by BerkeleyTrue** [comment]
@raisedadead What do you mean? We've had a few PR's fixing loop protect recently.

**Comment by raisedadead** [comment]
The do while loop seems to be in trouble.

It acts weird if there is a new line in between:
```js
do {
 my awesome code;
} while (condition);
```
The above will work while the below will fail.
```js

do {

 my awesome code;

} while (condition);
```

Refer:
https://github.com/FreeCodeCamp/FreeCodeCamp/issues/5162
https://github.com/FreeCodeCamp/FreeCodeCamp/issues/9652

**Comment by BerkeleyTrue** [comment]
> I think this is a good chance to have a look at an alternate better maintained library, the LP project is mostly dead

This sounds like hyperbole considering that the one issue with loop protect is do-while loops. (I think recur functions as well).

Remember we have thousands of challenges being submitted a day and loop protect is running on all of them. Not to mention that it is also running on JSBin. 

Yes, the issue's you listed have not been address in a long time but they are also effecting a very small percentage of users. For the most part, Loop Protect is as good as we are going to get without inventing a full JavaScript parser. (cough... AST's... cough...).

It is a proven technology and I'd rather see it our efforts put to fixing that edge cases then looking for alternatives at this moment and the near future.

**Comment by raisedadead** [comment]
Just tested this on the jsbin, and yes unfortunately the `do...while` problem exists (its quite weird that they catch it at times and then it breaks at times).

I completely agree that **we should** re-use the existing things, and fix them. 

Apologies, if I was misunderstood,its not just with do...while, but [multi line comments](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/7072) and [while loop with a callback](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/7105), earlier I just pointed the ones which I happened to visit recently.

My only concern was the [loop-protect](https://github.com/jsbin/loop-protect) repository is not active, perhaps even neglected, sure **if we come up** with a fix and patch our version that's super great and maybe make a pr as well on that repository. 

Having that said, yeah, we should NOT replace an entire library for the sake of some edge cases.

Will try and take a look on the bugs, soon. 

Maybe a good starting point would be what's different between the jsbin version and ours.
